exo-setup/src/app-setup.ls

```diff
@@ -23,36 +28,66 @@ class AppSetup extends EventEmitter
             location: service-data.location
             docker-image: service-data.docker_image
 
+    @_setup-services ~>
+      @_get-exocom-docker-config!
+      @_get-service-docker-config!
+      @_render-docker-compose!
+      @_setup-docker-images (err) ~>
+        | err => throw new Error err 
+        @logger.log role: 'exo-setup', text: 'setup complete' 
+
+
+  _setup-services: (done) ->
     setups = for service in @services
       if service.location
         new ServiceSetup role: service.role, logger: @logger, config: root: path.join(process.cwd!, service.location)
           ..on 'output', (data) ~> @emit 'output', data
     async.map-series setups, (-> &0.start &1), (err) ~>
       | err  =>  throw new Error err
+      done!
+
+
+  _get-exocom-docker-config: ->
+    exocom-docker-config =
+      exocom:
+        image: "originate/exocom:#{@app-config.bus.version}"
```

Comment:
It's probably better to do that all in one PR and keep this already massive PR as simple as possible.